hw3/02-url.js

```diff
@@ -11,29 +40,48 @@ const server = http.createServer((req, res) => {
 
   // use the URL interface to work with URLs
   // source: https://developer.mozilla.org/en-US/docs/Web/API/URL
-  let url = new URL(req.url, `http://${req.headers.host}`);
-
-  let getRoutes = () => {
+  const url = new URL(req.url, `http://${req.headers.host}`);
+  const urlPathName = url.pathname;
+  const getRoutes = () => {
     let result = '';
 
     routes.forEach(
-      (elem) => (result += `<li><a href="${elem}">${elem}</a></li>`)
+      (elem) => (result += `<li><a href="${elem}">${elem}</a></li>`),
     );
 
     return result;
   };
 
-  if (req.url === '/') {
-    let routeResults = getRoutes();
+  if (req.url === '/') { // Route for home page
+    const routeResults = getRoutes();
 
     res.writeHead(200, { 'Content-Type': 'text/html' });
-    res.write(`<h1>Exercise 02</h1>`);
+    res.write('<h1>Exercise 02</h1>');
 
     res.write(`<ul> ${routeResults} </ul>`);
+  } else if (urlPathName === '/attributes') { // Route for path attributes
+    const params = url.search;
```

Comment:
I see that you used the url api to retrieve the query string from the url. I did the same but used the url.searchParams() functions instead of url.search(). If you haven't I think it's totally worth checking out some other members and methods provided in the api!